serverless-functions/src/functions/features/realtime-voice-transcriptions/transcriptions-webhook.js

```diff
@@ -0,0 +1,118 @@
+
+const { OpenAI } = require('openai');
+const SyncOperations = require(Runtime.getFunctions()['common/twilio-wrappers/sync'].path);
+exports.handler = async function (context, event, callback) {
+    const openai = new OpenAI({
+        apiKey: context.OPENAI_API_KEY,
+    });
+    const response = new Twilio.Response();
+
+    response.appendHeader('Access-Control-Allow-Origin', '*');
+    response.appendHeader('Access-Control-Allow-Methods', 'OPTIONS POST GET');
+    response.appendHeader('Content-Type', 'application/json');
+    response.appendHeader('Access-Control-Allow-Headers', 'Content-Type');
+    // console.log(event)
+
+    const formatMessage = (message, direction) => ({
+        role: direction.includes('inbound_track') ? 'user' : 'assistant',
+        content: message,
+    });
+    switch (event.TranscriptionEvent) {
+        case 'transcription-started':
+            console.log('transcriptionEvent: transcription-started: ', event.CallSid);
+
+            //Create Sync Stream
+            const syncStreamResult = await SyncOperations.createStream({ context, name: `TRANSCRIPTION_${event.CallSid}` });
+            console.log('Sync Stream create: ', syncStreamResult);
+
+            // Create Sync map Item for call sid with TTL
+            const SyncMapResult = await SyncOperations.createMapItem({
+                context,
+                mapSid: context.MAP_CALL_LOG,
+                key: event.CallSid,
+                ttl: 84000,
+                data: { syncStream: syncStreamResult.stream.sid }
+            })
+            console.log('Sync Map Item create: ', SyncMapResult)
+            break;
+        case 'transcription-content':
+            if (event.Track == 'inbound_track') {
+                const transcript = JSON.parse(event.TranscriptionData).transcript;
+                console.log('transcription: user: ', transcript);
+                const syncStreamInboundData = {
+                    actor: 'Inbound',
+                    type: 'transcript',
+                    taranscriptionText: transcript,
+                }
+                const streamMessageInboundResult = await SyncOperations.createStreamMessage({ context, name: `TRANSCRIPTION_${event.CallSid}`, data: syncStreamInboundData });
+                console.log(streamMessageInboundResult);
+
+                const prompt = [];
+                prompt.push(formatMessage(transcript, event.Track));
+
+                prompt.push({
+                    role: 'system',
+                    content: `Always respond with a json array containing one or more objects with keys "suggestion" and "title", always complete both fields. You are a customer service agent assistant. Here is the call transcript so far, make a recommendation to the agent with the next best action or suggestion. Use the customers language where possible.`,
+                });
+                const result = await openai.chat.completions.create({
+                    model: 'gpt-3.5-turbo',
```

Comment:
Pass the model parameter in from .env so that we can upgrade it easily. 
Additionally, the model should be updated to gpt-4o